File: src/test/resources/org/jetbrains/research/ml/dataset/anonymizer/transformation/java/anonymization/data/classes_and_methods/out_4.java
public abstract class c1 {
    abstract public void c1_f1(int c1_f1_p1, int c1_f1_p2);
}

public class c2 extends c1 {
    @Override
    public void c1_f1(int c1_f1_p1, int c1_f1_p2) {

    }
}

public class c3 extends c2 {
    @Override
    public void c1_f1(int c1_f1_p1, int c1_f1_p2) {

    }

    abstract public void c3_f1(int c3_f1_p1, int c3_f1_p2);
}

public class c4 extends c3 {
    @Override
    public void c1_f1(int c1_f1_p1, int c1_f1_p2) {

    }

    @Override
    public void c3_f1(int c3_f1_p1, int c3_f1_p2) {

    }
}

File: src/test/resources/org/jetbrains/research/ml/dataset/anonymizer/transformations/anonymization/kotlin/data/classes_and_methods/out_4.kt
abstract class c1 {
    abstract fun c1_f1(c1_f1_p1: Int, c1_f1_p2: Int)
}

open class c2 : c1() {
    override fun c1_f1(c1_f1_p1: Int, c1_f1_p2: Int) {}
}

abstract class c3 : c2() {
    override fun c1_f1(c1_f1_p1: Int, c1_f1_p2: Int) {}
    abstract fun c3_f1(c3_f1_p1: Int, c3_f1_p2: Int)
}

class c4 : c3() {
    override fun c1_f1(c1_f1_p1: Int, c1_f1_p2: Int) {}
    override fun c3_f1(c3_f1_p1: Int, c3_f1_p2: Int) {}
}
